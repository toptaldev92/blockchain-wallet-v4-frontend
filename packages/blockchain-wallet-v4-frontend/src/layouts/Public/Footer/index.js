import React from 'react'
import styled from 'styled-components'

import { Badge } from 'blockchain-info-components'
import media from 'services/ResponsiveService'

const Wrapper = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  width: 100%;
  ${media.mobile`
    img {
      height: auto;
      width: 40%;
    }
  `}
`
const Bottom = styled.div`
  margin: 16px 0;
  > a {
    margin: 0 8px;
  }
`

const Footer = () => {
  return (
    <Wrapper>
      <Bottom>
        <Badge type='applestore' />
        <Badge type='googleplay' />
      </Bottom>
    </Wrapper>
  )
}

export default Footer
