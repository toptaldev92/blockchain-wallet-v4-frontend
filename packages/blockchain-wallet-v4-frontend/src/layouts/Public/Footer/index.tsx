import { Link } from 'blockchain-info-components'
import React from 'react'
import styled from 'styled-components'

const VersionWrapper = styled.span`
  margin-top: 4px;
  margin-left: 16px;
`

// @ts-ignore
const V = APP_VERSION

const Footer = () => (
  <VersionWrapper>
    <Link
      color="whiteFade600"
      href="https://github.com/blockchain/blockchain-wallet-v4-frontend/releases"
      size="14px"
      target="_blank"
      weight={500}
    >
      Version {V}
    </Link>
  </VersionWrapper>
)

export default Footer
