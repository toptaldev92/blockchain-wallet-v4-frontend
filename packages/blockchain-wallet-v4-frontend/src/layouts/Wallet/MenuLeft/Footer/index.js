import { actions, model } from 'data'
import { bindActionCreators } from 'redux'
import { Button, Icon, Link, Text } from 'blockchain-info-components'
import { connect } from 'react-redux'
import { FormattedHTMLMessage, FormattedMessage } from 'react-intl'
import { includes } from 'ramda'
import React from 'react'
import styled from 'styled-components'

const { ADS_EVENTS } = model.analytics

const Wrapper = styled.div`
  text-align: center;
`
const AdsButton = styled(Button)`
  margin: 8px auto 0px auto;
  transition: background 0.3s;
  line-height: normal;
  span > span:not(:first-child) {
    color: ${props => props.theme.blue600};
  }
  &:hover {
    span:first-child {
      color: ${props => props.theme.grey800};
    }
  }
`
const ButtonText = styled(Text)`
  display: flex;
  white-space: nowrap;
  align-items: center;
`
const ArrowIcon = styled(Icon)`
  margin-left: 4px;
  margin-top: 2px;
  font-size: 16px;
`
const SpadeIcon = styled(Icon)`
  margin-right: 8px;
`

const Footer = ({ actions, countryCode, adsBlacklist, adsUrl }) => {
  const isBlacklisted = includes(countryCode, adsBlacklist)
  return isBlacklisted ? null : (
    <Wrapper>
      <Text color='gray-3' size='12px' weight={500}>
        <FormattedMessage
          id='layouts.wallet.menuleft.footer.ad'
          defaultMessage='Ad by bitcoin.com'
        />
      </Text>
      <AdsButton
        height='48px'
        onClick={() => actions.logEvent(ADS_EVENTS.CLICK_AD)}
      >
        <Link href={adsUrl} rel='noopener noreferrer' target='_blank'>
          <ButtonText color='gray-4' size='14px' weight={500}>
            <SpadeIcon name='spade' color='gray4' />
            <FormattedHTMLMessage
              id='layouts.wallet.menuleft.footer.bitcoingames2'
              defaultMessage='<span>Bitcoin Games.</span> <span>Play Now</span>'
            />
            <ArrowIcon name='arrow-right' color='blue600' />
          </ButtonText>
        </Link>
      </AdsButton>
    </Wrapper>
  )
}

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.analytics, dispatch)
})

export default connect(
  null,
  mapDispatchToProps
)(Footer)
