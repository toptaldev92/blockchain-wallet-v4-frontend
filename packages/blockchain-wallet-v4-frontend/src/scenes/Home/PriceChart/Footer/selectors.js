import { createDeepEqualSelector } from 'services/ReselectHelper'
import { includes, path } from 'ramda'
import { selectors } from 'data'

export const getData = createDeepEqualSelector(
  [
    selectors.components.priceChart.getCoin,
    selectors.core.settings.getCountryCode,
    selectors.core.walletOptions.getCoinifyCountries,
    selectors.core.walletOptions.getSupportedCoins,
    selectors.modules.profile.isSilverOrAbove
  ],
  (
    coinTicker,
    countryCode,
    coinifyCountries,
    supportedCoins,
    isSilverOrAbove
  ) => {
    const coinName = path(['data', coinTicker, 'displayName'], supportedCoins)
    const isCoinifySupported = includes(countryCode.data, coinifyCountries.data)

    return {
      coinTicker,
      coinName,
      isCoinifySupported,
      isSilverOrAbove
    }
  }
)
