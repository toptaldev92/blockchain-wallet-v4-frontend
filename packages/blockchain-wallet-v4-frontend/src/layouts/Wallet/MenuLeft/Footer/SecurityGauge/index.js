import React from 'react'
import { compose, bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { actions } from 'data'
import { getData } from './selectors'
import { SecurityGauge } from 'blockchain-info-components'

class SecurityGaugeContainer extends React.PureComponent {
  render () {
    const { data } = this.props

    return data.cata({
      Success: value => <SecurityGauge score={value} />,
      Failure: message => <div />,
      Loading: () => <div />,
      NotAsked: () => <div />
    })
  }
}

const mapStateToProps = state => ({
  data: getData(state)
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.core.settings, dispatch)
})

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)

export default enhance(SecurityGaugeContainer)
