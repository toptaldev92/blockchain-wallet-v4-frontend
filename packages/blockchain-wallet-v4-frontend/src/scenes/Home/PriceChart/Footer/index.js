import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { actions } from 'data'
import { getData } from './selectors'
import Footer from './template'

class FooterContainer extends React.PureComponent {
  render () {
    return <Footer {...this.props} />
  }
}

const mapStateToProps = state => getData(state)

const mapDispatchToProps = dispatch => ({
  analyticsActions: bindActionCreators(actions.analytics, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FooterContainer)
