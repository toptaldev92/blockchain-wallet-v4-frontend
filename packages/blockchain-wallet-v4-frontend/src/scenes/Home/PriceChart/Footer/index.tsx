import { actions } from 'data'
import { bindActionCreators } from 'redux'
import { connect, ConnectedProps } from 'react-redux'
import { getData } from './selectors'
import Footer from './template'
import React from 'react'

class FooterContainer extends React.PureComponent<Props> {
  render () {
    const { cryptoCurrency, simpleBuyActions } = this.props
    return (
      <Footer
        handleBuy={() =>
          simpleBuyActions.showModal('PriceChart', cryptoCurrency)
        }
        {...this.props}
      />
    )
  }
}

const mapStateToProps = state => getData(state)

const mapDispatchToProps = dispatch => ({
  simpleBuyActions: bindActionCreators(actions.components.simpleBuy, dispatch)
})

const connector = connect(mapStateToProps, mapDispatchToProps)

type Props = ConnectedProps<typeof connector>

export default connector(FooterContainer)
