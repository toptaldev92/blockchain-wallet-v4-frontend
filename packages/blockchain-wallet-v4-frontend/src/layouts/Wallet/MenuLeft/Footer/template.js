import React from 'react'
import { FormattedMessage } from 'react-intl'
import { LinkContainer } from 'react-router-bootstrap'
import { Icon } from 'blockchain-info-components'
import SecurityGauge from './SecurityGauge'
import { Wrapper, MenuItem, SubMenu, SubMenuItem } from 'components/MenuLeft'

const Footer = props => {
  const { settingsOpened, userFlowSupported } = props.data

  return (
    <Wrapper>
      <LinkContainer to='/security-center' activeClassName='active'>
        <MenuItem data-e2e='securityCenterLink'>
          <Icon name='security' />
          <FormattedMessage
            id='layouts.wallet.menuleft.navigation.securitycenter'
            defaultMessage='Security Center'
          />
          <SecurityGauge />
        </MenuItem>
      </LinkContainer>
      <LinkContainer
        to='/settings/general'
        activeClassName='active'
        className={settingsOpened ? 'active' : ''}
      >
        <MenuItem data-e2e='settingsLink'>
          <Icon name='settings' />
          <FormattedMessage
            id='layouts.wallet.menuleft.navigation.settings'
            defaultMessage='Settings'
          />
        </MenuItem>
      </LinkContainer>
      {settingsOpened && (
        <SubMenu>
          <LinkContainer to='/settings/general' activeClassName='active'>
            <SubMenuItem data-e2e='settings_generalLink'>
              <FormattedMessage
                id='layouts.wallet.menuleft.navigation.general'
                defaultMessage='General'
              />
            </SubMenuItem>
          </LinkContainer>
          {userFlowSupported && (
            <LinkContainer to='/settings/profile' activeClassName='active'>
              <SubMenuItem data-e2e='settings_profileLink'>
                <FormattedMessage
                  id='layouts.wallet.menuleft.navigation.profile'
                  defaultMessage='Profile'
                />
              </SubMenuItem>
            </LinkContainer>
          )}
          <LinkContainer to='/settings/preferences' activeClassName='active'>
            <SubMenuItem data-e2e='settings_preferencesLink'>
              <FormattedMessage
                id='layouts.wallet.menuleft.navigation.preferences'
                defaultMessage='Preferences'
              />
            </SubMenuItem>
          </LinkContainer>
          <LinkContainer to='/settings/addresses/btc' activeClassName='active'>
            <SubMenuItem data-e2e='settings_walletsLink'>
              <FormattedMessage
                id='layouts.wallet.menuleft.navigation.addresses'
                defaultMessage='Wallets & Addresses'
              />
            </SubMenuItem>
          </LinkContainer>
        </SubMenu>
      )}
    </Wrapper>
  )
}

export default Footer
