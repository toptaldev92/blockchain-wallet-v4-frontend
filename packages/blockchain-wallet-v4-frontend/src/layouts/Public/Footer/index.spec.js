import { shallow } from 'enzyme'
import Footer from './index'
import React from 'react'
import toJson from 'enzyme-to-json'

jest.mock('components/DropdownLanguage', () => ({ DropdownLanguage: '' }))
jest.mock('blockchain-info-components', () => ({}))

describe('Public Footer Component', () => {
  it('should match snapshot', () => {
    const component = shallow(<Footer />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
