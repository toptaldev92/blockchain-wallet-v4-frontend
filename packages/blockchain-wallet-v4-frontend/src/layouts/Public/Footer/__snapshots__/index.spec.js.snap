// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Public Footer Component should match snapshot 1`] = `
<styled.footer>
  <styled.div>
    <Component
      type="applestore"
    />
    <Component
      type="googleplay"
    />
  </styled.div>
</styled.footer>
`;
